browser/plugins/three_material_modifier.plugin.js

```diff
@@ -0,0 +1,131 @@
+(function() {
+	var ThreeMaterialModifier = E2.plugins.three_material_modifier = function(core) {
+		Plugin.apply(this, arguments)
+
+		this.input_slots = [
+			{
+				name: 'material',
+				dt: core.datatypes.MATERIAL,
+				array: true,
+				def: [new THREE.MeshBasicMaterial({name: 'default material', color: 0x00FF00})]
+			}
+		]
+
+		this.output_slots = [
+			{
+				name: 'material',
+				dt: core.datatypes.MATERIAL,
+				array: true
+			}
+		]
+	}
+
+	ThreeMaterialModifier.prototype = Object.create(Plugin.prototype)
+
+	ThreeMaterialModifier.prototype.reset = function() {
+		this.in_material_array = []
+		this.out_material_array = []
+		this.material_override = {}
+
+		this.material_slots_dirty = false
+		this.materials_dirty = false
+	}
+
+	ThreeMaterialModifier.prototype.update_input = function(slot, data) {
+		if (!slot.uid && slot.index == 0) {
+			if (data) {
+				if (this.in_material_array.length == data.length) {
```

Comment:
can't early out on !data as we need to remove stuff from the array in the else block